AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginLazyLoadTests/LL12/HasOneParentChild/AWSDataStoreLazyLoadHasOneTests.swift

```diff
@@ -76,38 +102,169 @@ class AWSDataStoreLazyLoadHasOneTests: AWSDataStoreLazyLoadBaseTest {
+        do {
+            _ = try await Amplify.API.mutate(request: createRequest)
+        } catch {
+            XCTFail("Failed to send mutation request \(error)")
         }
 
+        await waitForExpectations([mutationEventReceived], timeout: 60)
+        mutationEvents.cancel()
+    }
+
+    /*
+     - Given: DataStore is cleared
+     - When:
+        - Configured with `HashOneModels`
+        - Create a new `HasOneChild` instance
+        - Delete the `HasOneChild` instance
+     - Then: Successfully deleted and synced to remote
+     */
+    func testDeleteHasOneChild_success() async throws {
+        await setup(withModels: HasOneModels())
+        let child = HasOneChild()
+        let savedChild = try await saveAndWaitForSync(child)
+        try await assertModelExists(savedChild)
+
+        try await deleteAndWaitForSync(savedChild)
+        try await assertModelDoesNotExist(savedChild)
+    }
+
+    /*
+     - Given: DataStore is cleared
+     - When:
+        - Configured with `HashOneModels`
+        - Create a new `HasOneParent` instance without `HasOneChild` association
+        - Delete the `HasOneParent` instance
+     - Then: Successfully deleted and synced to remote
+     */
+    func testDeleteHasOneParent_withoutChild_success() async throws {
+        await setup(withModels: HasOneModels())
+        let parent = HasOneParent()
+        let savedParent = try await saveAndWaitForSync(parent)
+        try await assertModelExists(savedParent)
+
+        try await deleteAndWaitForSync(savedParent)
+        try await assertModelDoesNotExist(savedParent)
+    }
+
+    /*
+     - Given: DataStore is cleared
+     - When:
+        - Configured with `HashOneModels`
+        - Create a new `HasOneParent` instance with `HasOneChild` association
+        - Delete the `HasOneParent` instance
+     - Then: Successfully deleted and synced to remote
+     */
+    func testDeleteHasOneParent_withChild_success() async throws {
+        await setup(withModels: HasOneModels())
+        let child = HasOneChild()
+        let savedChild = try await saveAndWaitForSync(child)
+        let parent = HasOneParent(child: savedChild, hasOneParentChildId: savedChild.id)
+        let savedParent = try await saveAndWaitForSync(parent)
+
+        let queriedParent = try await query(for: savedParent)
+        XCTAssertEqual(queriedParent.hasOneParentChildId, savedChild.id)
+
+        try await deleteAndWaitForSync(savedParent)
+        try await assertModelDoesNotExist(savedParent)
+        try await assertModelExists(savedChild)
+    }
+
+    /*
+     - Given: DataStore is cleared
+     - When:
+        - Configured with `HashOneModels`
+        - Create a new `HasOneChild` instance and associated to a `HasOneParent` instance
+        - Delete the `HasOneChild` instance
+     - Then: Successfully deleted and synced to remote
+     */
+    func testDeleteHasOneChild_withParent_success() async throws {
+        await setup(withModels: HasOneModels())
```

Comment:
nit: Should we move `await setup(withModels: HasOneModels())` to a setup function ?